chef/cookbooks/provisioner/libraries/repositories.rb

```diff
@@ -90,36 +92,40 @@ def inspect_repos(node)
           hae_available = false
           storage_available = false
 
-          %w(11.3 11.4 12.0 12.1).each do |version|
-            repos.merge! suse_get_repos_from_attributes(node,"suse",version)
-
-            # Common optional repos (regardless of cloud vs. storage)
-            suse_optional_repos(version, :common).each do |name|
-              repos[name] ||= Mash.new
-              next unless repos[name][:url].nil?
-              common_available ||= File.exist?("#{node[:provisioner][:root]}/suse-#{version}/repos/#{name}/repodata/repomd.xml")
-            end
+          node[:provisioner][:supported_oses].each do |os, arches|
+            next unless os =~ /^suse-/
+            version = os.gsub(/^suse-/, "")
+            arches.each do |arch, params|
+              repos.merge! suse_get_repos_from_attributes(node, "suse", version, arch)
+
+              # Common optional repos (regardless of cloud vs. storage)
+              suse_optional_repos(version, :common).each do |name|
+                repos[name] ||= Mash.new
+                next unless repos[name][:url].nil?
+                common_available ||= File.exist?("#{node[:provisioner][:root]}/suse-#{version}/#{arch}/repos/#{name}/repodata/repomd.xml")
+              end
 
-            # For cloud
-            suse_optional_repos(version, :cloud).each do |name|
-              repos[name] ||= Mash.new
-              next unless repos[name][:url].nil?
-              cloud_available ||= File.exist?("#{node[:provisioner][:root]}/suse-#{version}/repos/#{name}/repodata/repomd.xml") ||
-                                File.exist?("#{node[:provisioner][:root]}/suse-#{version}/repos/#{name}/suse/repodata/repomd.xml")
-            end
+              # For cloud
+              suse_optional_repos(version, :cloud).each do |name|
+                repos[name] ||= Mash.new
+                next unless repos[name][:url].nil?
+                cloud_available ||= File.exist?("#{node[:provisioner][:root]}/suse-#{version}/#{arch}/repos/#{name}/repodata/repomd.xml") ||
+                                  File.exist?("#{node[:provisioner][:root]}/suse-#{version}/#{arch}/repos/#{name}/suse/repodata/repomd.xml")
```

Comment:
Style/MultilineOperationIndentation: Use 2 (not 18) spaces for indenting an expression in an assignment spanning multiple lines.